File: src/main/java/com/atlaspharmacy/atlaspharmacy/generalities/model/Address.java
package com.atlaspharmacy.atlaspharmacy.generalities.model;

import javax.persistence.*;

@Entity
@Table(name = "addresses")
public class Address {
    @Id
    private int id;
    private String street;
    private int number;
    @ManyToOne(fetch = FetchType.LAZY, cascade = CascadeType.ALL)
    private City city;

    public Address() {}
    public Address(String street, int number, City city) {
        this.street = street;
        this.number = number;
        this.city = city;
    }



}

File: frontend/src/app/model/address/address.ts
import { City } from "./city";
import { Coordinates } from "./coordinates";
import { State } from "./state";

export class Address{
    public id: Number;
    public coordinates : Coordinates;
    public state : State;
    public city : City;
    public street : String;


    constructor(id:Number,street:String,city:City, coords : Coordinates, state : State){
        this.id = id;
        this.street = street;
        this.city = city;
        this.coordinates = coords;
        this.state = state;
    }
}
